### PR TITLE
feature: support one-to-many model for the leaf node feature

### PR DESCRIPTION
**Pull request by qiuming520**
#### What type of PR is this?
support one-to-many model for the leaf node feature

/kind feature 
#### Which issue(s) does this PR fix?

Fixes #https://github.com/kosmos-io/kosmos/issues/285
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by wuyingjun-lucky** [comment]
@qiuming520  Are you ready ?

**Comment by qiuming520** [comment]
> @qiuming520 Are you ready ?
yes~now do some tests

File: .golangci.yml [hunk 1]
```diff
@@ -6,7 +6,6 @@ linters:
   enable:
     - whitespace
     - bodyclose
-    - dupl
     - errcheck
```
Comment (duanmengkk) [inline .golangci.yml]: Why should delete this?

File: pkg/clustertree/cluster-manager/utils/leaf_resource_manager.go [hunk 2]
```diff
@@ -24,14 +24,15 @@ var (
 type LeafMode int
 
 const (
-	ALL  LeafMode = 0
-	Node LeafMode = 1
-	// Party LeafMode = 2
+	ALL   LeafMode = 0
```
Comment (duanmengkk) [inline pkg/clustertree/cluster-manager/utils/leaf_resource_manager.go]: use iota

File: test/e2e/leaf_node_test.go [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+			}
+
+			err := framework.DeleteClusters(hostClusterLinkClient, one2Party.Name)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+			if len(partyNodeNames) > 0 {
+				for _, node := range partyNodeNames {
+					err = framework.DeleteNode(hostKubeClient, node)
+					if err != nil && !apierrors.IsNotFound(err) {
+						gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
```
Comment (duanmengkk) [inline test/e2e/leaf_node_test.go:263]: We should check some conditions in e2e rather than check err is not occured

File: .golangci.yml (hunk 1 above)
Comment (qiuming520) [reply .golangci.yml]: In e2e, it's easy to have similar code structures and different logics

File: pkg/clustertree/cluster-manager/utils/leaf_resource_manager.go (hunk 2 above)
Comment (qiuming520) [reply pkg/clustertree/cluster-manager/utils/leaf_resource_manager.go]: Good idea ！！！

File: test/e2e/leaf_node_test.go (hunk 3 above)
Comment (qiuming520) [reply test/e2e/leaf_node_test.go:263]: Yes, specific condition cases will be added in other PRs

**Comment by duanmengkk** [comment]
the e2e test case is updated,fetch the new code and push again @qiuming520

**Pull request merged**